style_guide.md

```diff
@@ -0,0 +1,270 @@
+2. Constant variables
+3. State variables
+4. Type declarations
+5. Events
+6. Modifiers
+7. Functions
+
+Functions and declarations should be grouped according to their visibility. The exception to this is that an alternative grouping is sensible to reflect context, i.e. functions are related in some manner.
+
+#### Multiple contracts in the same file
+* Each smart contract should be in its own file.
+
+#### Import Statements
+* Import statements must always be placed at the top of the file.
+* They should be explicitly named import statements, not file level imports.
+```solidity
+Import { Something } from “./MySolidityFile.sol”;
+```
+
+## Naming Conventions
+Keep names as short as possible, wihtout losing precision or clarity in communicating its intent.[^1]
+
+#### Contract and Library Names
+* Contracts and libraries must be named using the CapWords style. 
+* Examples: `SimpleToken`, `SmartVault`, `Ownable`.
+
+#### Struct Names
+* Structs must be named using the CapWords style. 
+* Examples: `MyCoin`, `Position`, `PositionXY`.
+
+#### Event Names
+* Events must be named using the CapWords style. 
+* Event names will typically be singular.
+* Examples: `Deposit`, `Transfer`, `Approval`.
+
+#### Function Names
+* Functions other than constructors must use mixedCase. 
+  * `getBalance`, `transfer`, `verifyOwner`. 
+* Private or internal functions should use _underscoreMixedCase. 
+  * `_calculateBalance` , `_doTransfer`.
+
+#### Local and State Variable Names
+- Use mixedCase
+  * `totalSupply`, `balancesOf`, `creatorAddress`.
+- Variable names that refer to contracts are in the form: 
+  * `daiToken`, `yvDaiToken`, `sharesToken`.
+- Token amounts are to be referred as the contract variable, without `Token`, plus a word denoting the action undetaken with the token. In/Out will be the preferred words to denote tokens being transferred into/out of the contract. 
+  * `daiIn`, `daiOut`, `sharesMinted`, `yvDaiReceived`.
+
+#### Constant Names
+* Constants must be named with all capital letters with underscores separating words. 
+* Examples: `MAX_BLOCKS`, `CALLBACK_SUCCESS`, `CONTRACT_VERSION`.
+
+#### Mapping Names
+* Mappings should be mixed case with names that are plural. 
+* Examples: `deposits`, `balances`, `registeredAddresses`.
+
+#### Modifier Names
+* Use mixedCase. 
+* Examples: `onlyOwner`, `onlyAfter`, `onlyDuringThePreSale`.
+
+#### Enums
+* Enums, in the style of simple type declarations, must be named using the CapWords style. 
+* Examples: `TokenGroup`, `Frame`, `HashStyle`.
+
+
+## Code Comments
+
+#### NatSpec
+* Add NatSpec in a sensible fashion for contracts, functions and state variables[^2].
+
+#### In-line commenting
+* Every line of assembly should be clearly documented, for example, unchecked.
+* Useful for clarifying anything that is surprising, complicated, or risky.
+> Note: if you need to add a comment to explain what is that you are doing, take a moment to consider if you should do it differently.
+
+*** 
+<br>
```

Comment:
?

Reply:
The note is in reference to a comment Alberto made in slack. The takeaway being that if you find yourself excessively commenting to explain everything out in function, perhaps its a sign that it should be simplified (broken into parts, lesser abstraction, etc). 

The implicit understanding being that one should code in a manner that the code itself is readable and can stand mostly on its own legs with minimal support from comments, when it comes to comprehension.